app/services/word_watcher.rb

```diff
@@ -64,6 +64,27 @@ def self.word_matcher_regexp_key(action)
     "watched-words-list:#{action}"
   end
 
+  def self.censor(html)
+    regexp = WordWatcher.word_matcher_regexp(:censor)
+    return html if regexp.blank?
+
+    doc = Nokogiri::HTML5::fragment(html)
+    doc.traverse do |node|
+      if node.text?
+        node.content = node.content.gsub(regexp) do |match|
```

Comment:
What do you mean by "bad" and "a lot"? The replace is fast even when the match is very long. What can be expensive is the matching and there is no way to set a timeout for it.

Reply:
I meant to replicate the protection we added on the client side in #12967.

If there's a regexp that matches all words, or worse, characters, it could do a lot of replacements.

Reply:
That protection is only for replace probably because of recursive replacements (u -> you -> yoyou -> yoyoyou -> ...). We do not have this protection for censor watched words on the client side either. Would you like me to implement it there too?